From 75ddf0641cb511c735bc978982df9db540d94961 Mon Sep 17 00:00:00 2001
From: Yongzhi Chen <ychena@apache.org>
Date: Mon, 12 Jun 2017 16:31:12 -0400
Subject: [PATCH 1177/1363] CDH-55705: HIVE-16875: Query against view with
 partitioned child on HoS fails with privilege
 exception. (Yongzhi Chen, reviewed by Aihua Xu)

Conflicts:
	ql/src/java/org/apache/hadoop/hive/ql/parse/spark/GenSparkUtils.java
	ql/src/test/results/clientpositive/spark/auto_sortmerge_join_12.q.out
	ql/src/test/results/clientpositive/spark/bucket_map_join_spark4.q.out
	ql/src/test/results/clientpositive/spark/bucketmapjoin1.q.out
	ql/src/test/results/clientpositive/spark/bucketmapjoin2.q.out
	ql/src/test/results/clientpositive/spark/bucketmapjoin3.q.out
	ql/src/test/results/clientpositive/spark/bucketmapjoin4.q.out
	ql/src/test/results/clientpositive/spark/join26.q.out
	ql/src/test/results/clientpositive/spark/join32.q.out
	ql/src/test/results/clientpositive/spark/join32_lessSize.q.out
	ql/src/test/results/clientpositive/spark/join33.q.out
	ql/src/test/results/clientpositive/spark/louter_join_ppr.q.out
	ql/src/test/results/clientpositive/spark/mapjoin_mapjoin.q.out
	ql/src/test/results/clientpositive/spark/optimize_nullscan.q.out
	ql/src/test/results/clientpositive/spark/outer_join_ppr.q.out
	ql/src/test/results/clientpositive/spark/ppd_join_filter.q.out
	ql/src/test/results/clientpositive/spark/router_join_ppr.q.out
	ql/src/test/results/clientpositive/spark/smb_mapjoin_15.q.out
	ql/src/test/results/clientpositive/spark/union22.q.out

Change-Id: Ib0be164bbdbb5a677c540cef893a4a2204ff2bf9
---
 .../test/resources/testconfiguration.properties    |    3 +-
 .../hadoop/hive/ql/parse/spark/GenSparkUtils.java  |   20 +++++--
 .../clientpositive/explaindenpendencydiffengs.q    |   10 ++++
 .../explaindenpendencydiffengs.q.out               |   50 +++++++++++++++++
 .../spark/explaindenpendencydiffengs.q.out         |   50 +++++++++++++++++
 .../clientpositive/spark/groupby_sort_1_23.q.out   |   26 ++++-----
 .../spark/groupby_sort_skew_1_23.q.out             |   26 ++++-----
 .../test/results/clientpositive/spark/join32.q.out |    6 +-
 .../clientpositive/spark/join32_lessSize.q.out     |   26 ++++-----
 .../test/results/clientpositive/spark/join33.q.out |    6 +-
 .../test/results/clientpositive/spark/join34.q.out |    4 +-
 .../test/results/clientpositive/spark/join35.q.out |    4 +-
 .../clientpositive/spark/mapjoin_mapjoin.q.out     |   12 ++--
 .../clientpositive/spark/optimize_nullscan.q.out   |   58 ++++++++++----------
 .../clientpositive/spark/ppd_join_filter.q.out     |    8 +--
 .../clientpositive/spark/transform_ppr1.q.out      |    8 +--
 .../clientpositive/spark/transform_ppr2.q.out      |    4 +-
 .../results/clientpositive/spark/union22.q.out     |   12 ++--
 .../results/clientpositive/spark/union24.q.out     |   38 ++++++-------
 .../results/clientpositive/spark/union_ppr.q.out   |    8 +--
 20 files changed, 250 insertions(+), 129 deletions(-)
 create mode 100644 ql/src/test/queries/clientpositive/explaindenpendencydiffengs.q
 create mode 100644 ql/src/test/results/clientpositive/explaindenpendencydiffengs.q.out
 create mode 100644 ql/src/test/results/clientpositive/spark/explaindenpendencydiffengs.q.out

diff --git a/itests/src/test/resources/testconfiguration.properties b/itests/src/test/resources/testconfiguration.properties
index ec32908..54cd78c 100644
--- a/itests/src/test/resources/testconfiguration.properties
+++ b/itests/src/test/resources/testconfiguration.properties
@@ -474,6 +474,7 @@ spark.query.files=add_part_multiple.q, \
   escape_distributeby1.q, \
   escape_orderby1.q, \
   escape_sortby1.q, \
+  explaindenpendencydiffengs.q, \
   filter_join_breaktask.q, \
   filter_join_breaktask2.q, \
   groupby1.q, \
@@ -1531,4 +1532,4 @@ hdfs.blobstore.query.files=add_part_exist.q,\
   union_top_level.q,\
   union_view.q,\
   unionall_join_nullconstant.q,\
-  view.q
\ No newline at end of file
+  view.q
diff --git a/ql/src/java/org/apache/hadoop/hive/ql/parse/spark/GenSparkUtils.java b/ql/src/java/org/apache/hadoop/hive/ql/parse/spark/GenSparkUtils.java
index e27ce0d..7ef036b 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/parse/spark/GenSparkUtils.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/parse/spark/GenSparkUtils.java
@@ -154,10 +154,20 @@ public MapWork createMapWork(GenSparkProcContext context, Operator<?> root,
     Preconditions.checkArgument(root instanceof TableScanOperator,
       "AssertionError: expected root to be an instance of TableScanOperator, but was "
       + root.getClass().getName());
-    String alias = ((TableScanOperator) root).getConf().getAlias();
-
+    String alias_id = null;
+    if (context.parseContext != null && context.parseContext.getTopOps() != null) {
+      for (String currentAliasID : context.parseContext.getTopOps().keySet()) {
+        Operator<? extends OperatorDesc> currOp = context.parseContext.getTopOps().get(currentAliasID);
+        if (currOp == root) {
+          alias_id = currentAliasID;
+          break;
+        }
+      }
+    }
+    if (alias_id == null)
+      alias_id = ((TableScanOperator) root).getConf().getAlias();
     if (!deferSetup) {
-      setupMapWork(mapWork, context, partitions, root, alias);
+      setupMapWork(mapWork, context, partitions, root, alias_id);
     }
 
     // add new item to the Spark work
@@ -169,10 +179,10 @@ public MapWork createMapWork(GenSparkProcContext context, Operator<?> root,
   // this method's main use is to help unit testing this class
   protected void setupMapWork(MapWork mapWork, GenSparkProcContext context,
       PrunedPartitionList partitions, Operator<? extends OperatorDesc> root,
-      String alias) throws SemanticException {
+      String alias_id) throws SemanticException {
     // All the setup is done in GenMapRedUtils
     GenMapRedUtils.setMapWork(mapWork, context.parseContext,
-        context.inputs, partitions, root, alias, context.conf, false);
+        context.inputs, partitions, root, alias_id, context.conf, false);
   }
 
   private void collectOperators(Operator<?> op, List<Operator<?>> opList) {
diff --git a/ql/src/test/queries/clientpositive/explaindenpendencydiffengs.q b/ql/src/test/queries/clientpositive/explaindenpendencydiffengs.q
new file mode 100644
index 0000000..38f9aff
--- /dev/null
+++ b/ql/src/test/queries/clientpositive/explaindenpendencydiffengs.q
@@ -0,0 +1,10 @@
+drop table if exists dependtest;
+create table dependtest (a string) partitioned by (b int);
+insert into table dependtest partition (b=1) values ("hello");
+create view viewtest as select * from dependtest where b = 1;
+set hive.cbo.enable=false;
+set hive.compute.query.using.stats=false;
+explain dependency select count(*) from viewtest ;
+drop view viewtest;
+drop table dependtest;
+
diff --git a/ql/src/test/results/clientpositive/explaindenpendencydiffengs.q.out b/ql/src/test/results/clientpositive/explaindenpendencydiffengs.q.out
new file mode 100644
index 0000000..1456097
--- /dev/null
+++ b/ql/src/test/results/clientpositive/explaindenpendencydiffengs.q.out
@@ -0,0 +1,50 @@
+PREHOOK: query: drop table if exists dependtest
+PREHOOK: type: DROPTABLE
+POSTHOOK: query: drop table if exists dependtest
+POSTHOOK: type: DROPTABLE
+PREHOOK: query: create table dependtest (a string) partitioned by (b int)
+PREHOOK: type: CREATETABLE
+PREHOOK: Output: database:default
+PREHOOK: Output: default@dependtest
+POSTHOOK: query: create table dependtest (a string) partitioned by (b int)
+POSTHOOK: type: CREATETABLE
+POSTHOOK: Output: database:default
+POSTHOOK: Output: default@dependtest
+PREHOOK: query: insert into table dependtest partition (b=1) values ("hello")
+PREHOOK: type: QUERY
+PREHOOK: Output: default@dependtest@b=1
+POSTHOOK: query: insert into table dependtest partition (b=1) values ("hello")
+POSTHOOK: type: QUERY
+POSTHOOK: Output: default@dependtest@b=1
+POSTHOOK: Lineage: dependtest PARTITION(b=1).a SIMPLE [(values__tmp__table__1)values__tmp__table__1.FieldSchema(name:tmp_values_col1, type:string, comment:), ]
+PREHOOK: query: create view viewtest as select * from dependtest where b = 1
+PREHOOK: type: CREATEVIEW
+PREHOOK: Input: default@dependtest
+PREHOOK: Output: database:default
+PREHOOK: Output: default@viewtest
+POSTHOOK: query: create view viewtest as select * from dependtest where b = 1
+POSTHOOK: type: CREATEVIEW
+POSTHOOK: Input: default@dependtest
+POSTHOOK: Output: database:default
+POSTHOOK: Output: default@viewtest
+PREHOOK: query: explain dependency select count(*) from viewtest
+PREHOOK: type: QUERY
+POSTHOOK: query: explain dependency select count(*) from viewtest
+POSTHOOK: type: QUERY
+{"input_partitions":[{"partitionParents":"[default@viewtest]","partitionName":"default@dependtest@b=1"}],"input_tables":[{"tablename":"default@viewtest","tabletype":"VIRTUAL_VIEW"},{"tablename":"default@dependtest","tabletype":"MANAGED_TABLE","tableParents":"[default@viewtest]"}]}
+PREHOOK: query: drop view viewtest
+PREHOOK: type: DROPVIEW
+PREHOOK: Input: default@viewtest
+PREHOOK: Output: default@viewtest
+POSTHOOK: query: drop view viewtest
+POSTHOOK: type: DROPVIEW
+POSTHOOK: Input: default@viewtest
+POSTHOOK: Output: default@viewtest
+PREHOOK: query: drop table dependtest
+PREHOOK: type: DROPTABLE
+PREHOOK: Input: default@dependtest
+PREHOOK: Output: default@dependtest
+POSTHOOK: query: drop table dependtest
+POSTHOOK: type: DROPTABLE
+POSTHOOK: Input: default@dependtest
+POSTHOOK: Output: default@dependtest
diff --git a/ql/src/test/results/clientpositive/spark/explaindenpendencydiffengs.q.out b/ql/src/test/results/clientpositive/spark/explaindenpendencydiffengs.q.out
new file mode 100644
index 0000000..1456097
--- /dev/null
+++ b/ql/src/test/results/clientpositive/spark/explaindenpendencydiffengs.q.out
@@ -0,0 +1,50 @@
+PREHOOK: query: drop table if exists dependtest
+PREHOOK: type: DROPTABLE
+POSTHOOK: query: drop table if exists dependtest
+POSTHOOK: type: DROPTABLE
+PREHOOK: query: create table dependtest (a string) partitioned by (b int)
+PREHOOK: type: CREATETABLE
+PREHOOK: Output: database:default
+PREHOOK: Output: default@dependtest
+POSTHOOK: query: create table dependtest (a string) partitioned by (b int)
+POSTHOOK: type: CREATETABLE
+POSTHOOK: Output: database:default
+POSTHOOK: Output: default@dependtest
+PREHOOK: query: insert into table dependtest partition (b=1) values ("hello")
+PREHOOK: type: QUERY
+PREHOOK: Output: default@dependtest@b=1
+POSTHOOK: query: insert into table dependtest partition (b=1) values ("hello")
+POSTHOOK: type: QUERY
+POSTHOOK: Output: default@dependtest@b=1
+POSTHOOK: Lineage: dependtest PARTITION(b=1).a SIMPLE [(values__tmp__table__1)values__tmp__table__1.FieldSchema(name:tmp_values_col1, type:string, comment:), ]
+PREHOOK: query: create view viewtest as select * from dependtest where b = 1
+PREHOOK: type: CREATEVIEW
+PREHOOK: Input: default@dependtest
+PREHOOK: Output: database:default
+PREHOOK: Output: default@viewtest
+POSTHOOK: query: create view viewtest as select * from dependtest where b = 1
+POSTHOOK: type: CREATEVIEW
+POSTHOOK: Input: default@dependtest
+POSTHOOK: Output: database:default
+POSTHOOK: Output: default@viewtest
+PREHOOK: query: explain dependency select count(*) from viewtest
+PREHOOK: type: QUERY
+POSTHOOK: query: explain dependency select count(*) from viewtest
+POSTHOOK: type: QUERY
+{"input_partitions":[{"partitionParents":"[default@viewtest]","partitionName":"default@dependtest@b=1"}],"input_tables":[{"tablename":"default@viewtest","tabletype":"VIRTUAL_VIEW"},{"tablename":"default@dependtest","tabletype":"MANAGED_TABLE","tableParents":"[default@viewtest]"}]}
+PREHOOK: query: drop view viewtest
+PREHOOK: type: DROPVIEW
+PREHOOK: Input: default@viewtest
+PREHOOK: Output: default@viewtest
+POSTHOOK: query: drop view viewtest
+POSTHOOK: type: DROPVIEW
+POSTHOOK: Input: default@viewtest
+POSTHOOK: Output: default@viewtest
+PREHOOK: query: drop table dependtest
+PREHOOK: type: DROPTABLE
+PREHOOK: Input: default@dependtest
+PREHOOK: Output: default@dependtest
+POSTHOOK: query: drop table dependtest
+POSTHOOK: type: DROPTABLE
+POSTHOOK: Input: default@dependtest
+POSTHOOK: Output: default@dependtest
diff --git a/ql/src/test/results/clientpositive/spark/groupby_sort_1_23.q.out b/ql/src/test/results/clientpositive/spark/groupby_sort_1_23.q.out
index 027e645..e71c3e9 100644
--- a/ql/src/test/results/clientpositive/spark/groupby_sort_1_23.q.out
+++ b/ql/src/test/results/clientpositive/spark/groupby_sort_1_23.q.out
@@ -617,7 +617,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
 
   Stage: Stage-0
     Move Operator
@@ -838,7 +838,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
 
   Stage: Stage-0
     Move Operator
@@ -1725,7 +1725,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
         Reducer 2 
             Needs Tagging: false
             Reduce Operator Tree:
@@ -2017,7 +2017,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery1:subq1-subquery1:t1]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -2121,7 +2121,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery2:subq1-subquery2:t1]
 
   Stage: Stage-0
     Move Operator
@@ -2391,7 +2391,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery1:subq1-subquery1:t1]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -2468,7 +2468,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery2:subq1-subquery2:t1]
         Reducer 3 
             Needs Tagging: false
             Reduce Operator Tree:
@@ -2770,7 +2770,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -2847,7 +2847,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq2:t1]
         Reducer 2 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -3133,7 +3133,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -3210,7 +3210,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq2:t1]
         Reducer 2 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -4123,7 +4123,7 @@ STAGE PLANS:
                     name: default.t2
                   name: default.t2
             Truncated Path -> Alias:
-              /t2 [t2]
+              /t2 [subq:t2]
 
   Stage: Stage-0
     Move Operator
@@ -4394,7 +4394,7 @@ STAGE PLANS:
                     name: default.t2
                   name: default.t2
             Truncated Path -> Alias:
-              /t2 [t2]
+              /t2 [subq2:subq:t2]
 
   Stage: Stage-0
     Move Operator
diff --git a/ql/src/test/results/clientpositive/spark/groupby_sort_skew_1_23.q.out b/ql/src/test/results/clientpositive/spark/groupby_sort_skew_1_23.q.out
index ee3dbe2..7d9c793 100644
--- a/ql/src/test/results/clientpositive/spark/groupby_sort_skew_1_23.q.out
+++ b/ql/src/test/results/clientpositive/spark/groupby_sort_skew_1_23.q.out
@@ -635,7 +635,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
 
   Stage: Stage-0
     Move Operator
@@ -856,7 +856,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
 
   Stage: Stage-0
     Move Operator
@@ -1780,7 +1780,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
         Reducer 2 
             Needs Tagging: false
             Reduce Operator Tree:
@@ -2089,7 +2089,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery1:subq1-subquery1:t1]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -2193,7 +2193,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery2:subq1-subquery2:t1]
 
   Stage: Stage-0
     Move Operator
@@ -2464,7 +2464,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery1:subq1-subquery1:t1]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -2541,7 +2541,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [null-subquery2:subq1-subquery2:t1]
         Reducer 3 
             Needs Tagging: false
             Reduce Operator Tree:
@@ -2860,7 +2860,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -2937,7 +2937,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq2:t1]
         Reducer 2 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -3224,7 +3224,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq1:t1]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -3301,7 +3301,7 @@ STAGE PLANS:
                     name: default.t1
                   name: default.t1
             Truncated Path -> Alias:
-              /t1 [t1]
+              /t1 [subq2:t1]
         Reducer 2 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -4249,7 +4249,7 @@ STAGE PLANS:
                     name: default.t2
                   name: default.t2
             Truncated Path -> Alias:
-              /t2 [t2]
+              /t2 [subq:t2]
 
   Stage: Stage-0
     Move Operator
@@ -4520,7 +4520,7 @@ STAGE PLANS:
                     name: default.t2
                   name: default.t2
             Truncated Path -> Alias:
-              /t2 [t2]
+              /t2 [subq2:subq:t2]
 
   Stage: Stage-0
     Move Operator
diff --git a/ql/src/test/results/clientpositive/spark/join32.q.out b/ql/src/test/results/clientpositive/spark/join32.q.out
index b1cc00d..f64fc98 100644
--- a/ql/src/test/results/clientpositive/spark/join32.q.out
+++ b/ql/src/test/results/clientpositive/spark/join32.q.out
@@ -171,7 +171,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [y]
+              /src [$hdt$_0:$hdt$_0:y]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -241,7 +241,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_2:x]
 
   Stage: Stage-1
     Spark
@@ -365,7 +365,7 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [z]
+              /srcpart/ds=2008-04-08/hr=11 [$hdt$_0:$hdt$_1:$hdt$_1:z]
 
   Stage: Stage-0
     Move Operator
diff --git a/ql/src/test/results/clientpositive/spark/join32_lessSize.q.out b/ql/src/test/results/clientpositive/spark/join32_lessSize.q.out
index ecba6dc..963f1eb 100644
--- a/ql/src/test/results/clientpositive/spark/join32_lessSize.q.out
+++ b/ql/src/test/results/clientpositive/spark/join32_lessSize.q.out
@@ -179,7 +179,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [y]
+              /src [$hdt$_0:$hdt$_0:y]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -249,7 +249,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_2:x]
 
   Stage: Stage-1
     Spark
@@ -373,7 +373,7 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [z]
+              /srcpart/ds=2008-04-08/hr=11 [$hdt$_0:$hdt$_1:$hdt$_1:z]
 
   Stage: Stage-0
     Move Operator
@@ -685,7 +685,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_2:$hdt$_2:x]
 
   Stage: Stage-3
     Spark
@@ -760,7 +760,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [w]
+              /src [$hdt$_0:$hdt$_0:w]
         Map 4 
             Map Operator Tree:
                 TableScan
@@ -841,7 +841,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_2:$hdt$_3:x]
 
   Stage: Stage-1
     Spark
@@ -968,7 +968,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [w]
+              /src [$hdt$_0:$hdt$_1:$hdt$_1:w]
 
   Stage: Stage-0
     Move Operator
@@ -1301,7 +1301,7 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [z]
+              /srcpart/ds=2008-04-08/hr=11 [$hdt$_0:$hdt$_0:z]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -1371,7 +1371,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_1:$hdt$_2:x]
 
   Stage: Stage-1
     Spark
@@ -1497,7 +1497,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [y]
+              /src [$hdt$_0:$hdt$_1:$hdt$_1:$hdt$_1:y]
 
   Stage: Stage-0
     Move Operator
@@ -1821,7 +1821,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [y]
+              /src [$hdt$_0:$hdt$_0:$hdt$_0:y]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -1893,7 +1893,7 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [z]
+              /srcpart/ds=2008-04-08/hr=11 [$hdt$_1:z]
 
   Stage: Stage-1
     Spark
@@ -2024,7 +2024,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_0:$hdt$_1:x]
 
   Stage: Stage-0
     Move Operator
diff --git a/ql/src/test/results/clientpositive/spark/join33.q.out b/ql/src/test/results/clientpositive/spark/join33.q.out
index b1cc00d..f64fc98 100644
--- a/ql/src/test/results/clientpositive/spark/join33.q.out
+++ b/ql/src/test/results/clientpositive/spark/join33.q.out
@@ -171,7 +171,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [y]
+              /src [$hdt$_0:$hdt$_0:y]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -241,7 +241,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [x]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_2:x]
 
   Stage: Stage-1
     Spark
@@ -365,7 +365,7 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [z]
+              /srcpart/ds=2008-04-08/hr=11 [$hdt$_0:$hdt$_1:$hdt$_1:z]
 
   Stage: Stage-0
     Move Operator
diff --git a/ql/src/test/results/clientpositive/spark/join34.q.out b/ql/src/test/results/clientpositive/spark/join34.q.out
index 209970a..15c50ba 100644
--- a/ql/src/test/results/clientpositive/spark/join34.q.out
+++ b/ql/src/test/results/clientpositive/spark/join34.q.out
@@ -212,7 +212,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [x]
+              /src [null-subquery1:subq1-subquery1:x]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -283,7 +283,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [x1]
+              /src [null-subquery2:subq1-subquery2:x1]
         Map 4 
             Map Operator Tree:
                 TableScan
diff --git a/ql/src/test/results/clientpositive/spark/join35.q.out b/ql/src/test/results/clientpositive/spark/join35.q.out
index 44ce9cb..8d11727 100644
--- a/ql/src/test/results/clientpositive/spark/join35.q.out
+++ b/ql/src/test/results/clientpositive/spark/join35.q.out
@@ -224,7 +224,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [x]
+              /src [null-subquery1:subq1-subquery1:x]
         Map 4 
             Map Operator Tree:
                 TableScan
@@ -297,7 +297,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [x1]
+              /src [null-subquery2:subq1-subquery2:x1]
         Map 6 
             Map Operator Tree:
                 TableScan
diff --git a/ql/src/test/results/clientpositive/spark/mapjoin_mapjoin.q.out b/ql/src/test/results/clientpositive/spark/mapjoin_mapjoin.q.out
index 314368e..29c2c92 100644
--- a/ql/src/test/results/clientpositive/spark/mapjoin_mapjoin.q.out
+++ b/ql/src/test/results/clientpositive/spark/mapjoin_mapjoin.q.out
@@ -126,7 +126,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [src]
+              /src [$hdt$_0:$hdt$_0:src]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -196,7 +196,7 @@ STAGE PLANS:
                     name: default.src1
                   name: default.src1
             Truncated Path -> Alias:
-              /src1 [src1]
+              /src1 [$hdt$_0:$hdt$_1:$hdt$_2:src1]
 
   Stage: Stage-1
     Spark
@@ -453,10 +453,10 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [srcpart]
-              /srcpart/ds=2008-04-08/hr=12 [srcpart]
-              /srcpart/ds=2008-04-09/hr=11 [srcpart]
-              /srcpart/ds=2008-04-09/hr=12 [srcpart]
+              /srcpart/ds=2008-04-08/hr=11 [$hdt$_0:$hdt$_1:$hdt$_1:srcpart]
+              /srcpart/ds=2008-04-08/hr=12 [$hdt$_0:$hdt$_1:$hdt$_1:srcpart]
+              /srcpart/ds=2008-04-09/hr=11 [$hdt$_0:$hdt$_1:$hdt$_1:srcpart]
+              /srcpart/ds=2008-04-09/hr=12 [$hdt$_0:$hdt$_1:$hdt$_1:srcpart]
 
   Stage: Stage-0
     Fetch Operator
diff --git a/ql/src/test/results/clientpositive/spark/optimize_nullscan.q.out b/ql/src/test/results/clientpositive/spark/optimize_nullscan.q.out
index 8a94672..e7e410d 100644
--- a/ql/src/test/results/clientpositive/spark/optimize_nullscan.q.out
+++ b/ql/src/test/results/clientpositive/spark/optimize_nullscan.q.out
@@ -243,7 +243,7 @@ STAGE PLANS:
                         tag: 0
                         auto parallelism: false
             Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [a:src]
             Path -> Partition:
               -mr-10003default.src{} 
                 Partition
@@ -289,7 +289,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [a:src]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -310,10 +310,10 @@ STAGE PLANS:
                         value expressions: _col0 (type: string)
                         auto parallelism: false
             Path -> Alias:
-              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [srcpart]
-              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [srcpart]
-              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [srcpart]
-              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [srcpart]
+              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [b:srcpart]
+              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [b:srcpart]
+              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [b:srcpart]
+              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [b:srcpart]
             Path -> Partition:
               -mr-10004default.srcpart{ds=2008-04-08, hr=11} 
                 Partition
@@ -496,10 +496,10 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [srcpart]
-              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [srcpart]
-              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [srcpart]
-              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [srcpart]
+              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [b:srcpart]
+              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [b:srcpart]
+              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [b:srcpart]
+              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [b:srcpart]
         Reducer 2 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -662,7 +662,7 @@ STAGE PLANS:
                         value expressions: _col0 (type: bigint)
                         auto parallelism: false
             Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [null-subquery1:_u1-subquery1:src]
             Path -> Partition:
               -mr-10003default.src{} 
                 Partition
@@ -708,7 +708,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [null-subquery1:_u1-subquery1:src]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -918,10 +918,10 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [srcpart]
-              /srcpart/ds=2008-04-08/hr=12 [srcpart]
-              /srcpart/ds=2008-04-09/hr=11 [srcpart]
-              /srcpart/ds=2008-04-09/hr=12 [srcpart]
+              /srcpart/ds=2008-04-08/hr=11 [null-subquery2:_u1-subquery2:srcpart]
+              /srcpart/ds=2008-04-08/hr=12 [null-subquery2:_u1-subquery2:srcpart]
+              /srcpart/ds=2008-04-09/hr=11 [null-subquery2:_u1-subquery2:srcpart]
+              /srcpart/ds=2008-04-09/hr=12 [null-subquery2:_u1-subquery2:srcpart]
         Reducer 2 
             Needs Tagging: false
             Reduce Operator Tree:
@@ -1095,7 +1095,7 @@ STAGE PLANS:
                         value expressions: _col0 (type: string)
                         auto parallelism: false
             Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [a:src]
             Path -> Partition:
               -mr-10003default.src{} 
                 Partition
@@ -1141,7 +1141,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [a:src]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -1162,10 +1162,10 @@ STAGE PLANS:
                         value expressions: _col0 (type: string)
                         auto parallelism: false
             Path -> Alias:
-              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [srcpart]
-              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [srcpart]
-              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [srcpart]
-              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [srcpart]
+              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [b:srcpart]
+              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [b:srcpart]
+              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [b:srcpart]
+              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [b:srcpart]
             Path -> Partition:
               -mr-10004default.srcpart{ds=2008-04-08, hr=11} 
                 Partition
@@ -1348,10 +1348,10 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [srcpart]
-              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [srcpart]
-              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [srcpart]
-              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [srcpart]
+              -mr-10004default.srcpart{ds=2008-04-08, hr=11} [b:srcpart]
+              -mr-10005default.srcpart{ds=2008-04-08, hr=12} [b:srcpart]
+              -mr-10006default.srcpart{ds=2008-04-09, hr=11} [b:srcpart]
+              -mr-10007default.srcpart{ds=2008-04-09, hr=12} [b:srcpart]
         Reducer 2 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -1585,7 +1585,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [src]
+              /src [null-subquery1:a-subquery1:src]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -1604,7 +1604,7 @@ STAGE PLANS:
                       tag: 0
                       auto parallelism: false
             Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [null-subquery2:a-subquery2:src]
             Path -> Partition:
               -mr-10003default.src{} 
                 Partition
@@ -1650,7 +1650,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              -mr-10003default.src{} [src]
+              -mr-10003default.src{} [null-subquery2:a-subquery2:src]
         Map 4 
         Reducer 3 
             Needs Tagging: true
diff --git a/ql/src/test/results/clientpositive/spark/ppd_join_filter.q.out b/ql/src/test/results/clientpositive/spark/ppd_join_filter.q.out
index 0cbd718..28c965d 100644
--- a/ql/src/test/results/clientpositive/spark/ppd_join_filter.q.out
+++ b/ql/src/test/results/clientpositive/spark/ppd_join_filter.q.out
@@ -203,7 +203,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [src]
+              /src [b:src]
         Map 4 
             Map Operator Tree:
                 TableScan
@@ -577,7 +577,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [src]
+              /src [b:src]
         Map 4 
             Map Operator Tree:
                 TableScan
@@ -951,7 +951,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [src]
+              /src [b:src]
         Map 4 
             Map Operator Tree:
                 TableScan
@@ -1325,7 +1325,7 @@ STAGE PLANS:
                     name: default.src
                   name: default.src
             Truncated Path -> Alias:
-              /src [src]
+              /src [b:src]
         Map 4 
             Map Operator Tree:
                 TableScan
diff --git a/ql/src/test/results/clientpositive/spark/transform_ppr1.q.out b/ql/src/test/results/clientpositive/spark/transform_ppr1.q.out
index 4d07be5..c5995f8 100644
--- a/ql/src/test/results/clientpositive/spark/transform_ppr1.q.out
+++ b/ql/src/test/results/clientpositive/spark/transform_ppr1.q.out
@@ -325,10 +325,10 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [src]
-              /srcpart/ds=2008-04-08/hr=12 [src]
-              /srcpart/ds=2008-04-09/hr=11 [src]
-              /srcpart/ds=2008-04-09/hr=12 [src]
+              /srcpart/ds=2008-04-08/hr=11 [tmap:src]
+              /srcpart/ds=2008-04-08/hr=12 [tmap:src]
+              /srcpart/ds=2008-04-09/hr=11 [tmap:src]
+              /srcpart/ds=2008-04-09/hr=12 [tmap:src]
         Reducer 2 
             Needs Tagging: false
             Reduce Operator Tree:
diff --git a/ql/src/test/results/clientpositive/spark/transform_ppr2.q.out b/ql/src/test/results/clientpositive/spark/transform_ppr2.q.out
index cc57601..dc2072a 100644
--- a/ql/src/test/results/clientpositive/spark/transform_ppr2.q.out
+++ b/ql/src/test/results/clientpositive/spark/transform_ppr2.q.out
@@ -235,8 +235,8 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [src]
-              /srcpart/ds=2008-04-08/hr=12 [src]
+              /srcpart/ds=2008-04-08/hr=11 [tmap:src]
+              /srcpart/ds=2008-04-08/hr=12 [tmap:src]
         Reducer 2 
             Needs Tagging: false
             Reduce Operator Tree:
diff --git a/ql/src/test/results/clientpositive/spark/union22.q.out b/ql/src/test/results/clientpositive/spark/union22.q.out
index 43d0d16..248c149 100644
--- a/ql/src/test/results/clientpositive/spark/union22.q.out
+++ b/ql/src/test/results/clientpositive/spark/union22.q.out
@@ -75,7 +75,7 @@ ABSTRACT SYNTAX TREE:
 TOK_QUERY
    TOK_FROM
       TOK_SUBQUERY
-         TOK_UNIONALL
+         TOK_UNION
             TOK_QUERY
                TOK_FROM
                   TOK_TABREF
@@ -296,7 +296,7 @@ STAGE PLANS:
                     name: default.dst_union22_delta
                   name: default.dst_union22_delta
             Truncated Path -> Alias:
-              /dst_union22_delta/ds=1 [dst_union22_delta]
+              /dst_union22_delta/ds=1 [null-subquery2:subq-subquery2:b:dst_union22_delta]
 
   Stage: Stage-1
     Spark
@@ -394,7 +394,7 @@ STAGE PLANS:
                     name: default.dst_union22_delta
                   name: default.dst_union22_delta
             Truncated Path -> Alias:
-              /dst_union22_delta/ds=1 [dst_union22_delta]
+              /dst_union22_delta/ds=1 [null-subquery1:subq-subquery1:dst_union22_delta]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -505,7 +505,7 @@ STAGE PLANS:
                     name: default.dst_union22
                   name: default.dst_union22
             Truncated Path -> Alias:
-              /dst_union22/ds=1 [a]
+              /dst_union22/ds=1 [null-subquery2:subq-subquery2:a]
 
   Stage: Stage-0
     Move Operator
@@ -573,8 +573,8 @@ POSTHOOK: Input: default@dst_union22_delta@ds=1
 POSTHOOK: Output: default@dst_union22@ds=2
 POSTHOOK: Lineage: dst_union22 PARTITION(ds=2).k1 EXPRESSION [(dst_union22_delta)dst_union22_delta.FieldSchema(name:k1, type:string, comment:null), (dst_union22)a.FieldSchema(name:k1, type:string, comment:null), ]
 POSTHOOK: Lineage: dst_union22 PARTITION(ds=2).k2 EXPRESSION [(dst_union22_delta)dst_union22_delta.FieldSchema(name:k2, type:string, comment:null), (dst_union22)a.FieldSchema(name:k2, type:string, comment:null), ]
-POSTHOOK: Lineage: dst_union22 PARTITION(ds=2).k3 EXPRESSION [(dst_union22_delta)dst_union22_delta.FieldSchema(name:k3, type:string, comment:null), (dst_union22_delta)dst_union22_delta.FieldSchema(name:k3, type:string, comment:null), ]
-POSTHOOK: Lineage: dst_union22 PARTITION(ds=2).k4 EXPRESSION [(dst_union22_delta)dst_union22_delta.FieldSchema(name:k4, type:string, comment:null), (dst_union22_delta)dst_union22_delta.FieldSchema(name:k4, type:string, comment:null), ]
+POSTHOOK: Lineage: dst_union22 PARTITION(ds=2).k3 EXPRESSION [(dst_union22_delta)dst_union22_delta.FieldSchema(name:k3, type:string, comment:null), ]
+POSTHOOK: Lineage: dst_union22 PARTITION(ds=2).k4 EXPRESSION [(dst_union22_delta)dst_union22_delta.FieldSchema(name:k4, type:string, comment:null), ]
 PREHOOK: query: select * from dst_union22 where ds = '2'
 PREHOOK: type: QUERY
 PREHOOK: Input: default@dst_union22
diff --git a/ql/src/test/results/clientpositive/spark/union24.q.out b/ql/src/test/results/clientpositive/spark/union24.q.out
index 9d8d47f..255ff3b 100644
--- a/ql/src/test/results/clientpositive/spark/union24.q.out
+++ b/ql/src/test/results/clientpositive/spark/union24.q.out
@@ -65,9 +65,9 @@ ABSTRACT SYNTAX TREE:
 TOK_QUERY
    TOK_FROM
       TOK_SUBQUERY
-         TOK_UNIONALL
-            TOK_UNIONALL
-               TOK_UNIONALL
+         TOK_UNION
+            TOK_UNION
+               TOK_UNION
                   TOK_QUERY
                      TOK_FROM
                         TOK_TABREF
@@ -269,7 +269,7 @@ STAGE PLANS:
                     name: default.src2
                   name: default.src2
             Truncated Path -> Alias:
-              /src2 [src2]
+              /src2 [null-subquery1-subquery1-subquery1:s-subquery1-subquery1-subquery1:src2]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -353,7 +353,7 @@ STAGE PLANS:
                     name: default.src3
                   name: default.src3
             Truncated Path -> Alias:
-              /src3 [src3]
+              /src3 [null-subquery1-subquery1-subquery2:s-subquery1-subquery1-subquery2:src3]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -437,7 +437,7 @@ STAGE PLANS:
                     name: default.src4
                   name: default.src4
             Truncated Path -> Alias:
-              /src4 [src4]
+              /src4 [null-subquery1-subquery2:s-subquery1-subquery2:src4]
         Map 4 
             Map Operator Tree:
                 TableScan
@@ -510,7 +510,7 @@ STAGE PLANS:
                     name: default.src5
                   name: default.src5
             Truncated Path -> Alias:
-              /src5 [src5]
+              /src5 [null-subquery2:s-subquery2:src5]
         Reducer 5 
             Needs Tagging: false
             Reduce Operator Tree:
@@ -625,8 +625,8 @@ ABSTRACT SYNTAX TREE:
 TOK_QUERY
    TOK_FROM
       TOK_SUBQUERY
-         TOK_UNIONALL
-            TOK_UNIONALL
+         TOK_UNION
+            TOK_UNION
                TOK_QUERY
                   TOK_FROM
                      TOK_TABREF
@@ -825,7 +825,7 @@ STAGE PLANS:
                     name: default.src2
                   name: default.src2
             Truncated Path -> Alias:
-              /src2 [src2]
+              /src2 [null-subquery1-subquery1:s-subquery1-subquery1:src2]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -909,7 +909,7 @@ STAGE PLANS:
                     name: default.src3
                   name: default.src3
             Truncated Path -> Alias:
-              /src3 [src3]
+              /src3 [null-subquery1-subquery2:s-subquery1-subquery2:src3]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -975,7 +975,7 @@ STAGE PLANS:
                     name: default.src4
                   name: default.src4
             Truncated Path -> Alias:
-              /src4 [a]
+              /src4 [null-subquery2:s-subquery2:a]
         Map 5 
             Map Operator Tree:
                 TableScan
@@ -1042,7 +1042,7 @@ STAGE PLANS:
                     name: default.src5
                   name: default.src5
             Truncated Path -> Alias:
-              /src5 [b]
+              /src5 [null-subquery2:s-subquery2:b]
         Reducer 4 
             Needs Tagging: true
             Reduce Operator Tree:
@@ -1150,8 +1150,8 @@ ABSTRACT SYNTAX TREE:
 TOK_QUERY
    TOK_FROM
       TOK_SUBQUERY
-         TOK_UNIONALL
-            TOK_UNIONALL
+         TOK_UNION
+            TOK_UNION
                TOK_QUERY
                   TOK_FROM
                      TOK_TABREF
@@ -1355,7 +1355,7 @@ STAGE PLANS:
                     name: default.src2
                   name: default.src2
             Truncated Path -> Alias:
-              /src2 [src2]
+              /src2 [null-subquery1-subquery1:s-subquery1-subquery1:src2]
         Map 2 
             Map Operator Tree:
                 TableScan
@@ -1439,7 +1439,7 @@ STAGE PLANS:
                     name: default.src3
                   name: default.src3
             Truncated Path -> Alias:
-              /src3 [src3]
+              /src3 [null-subquery1-subquery2:s-subquery1-subquery2:src3]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -1505,7 +1505,7 @@ STAGE PLANS:
                     name: default.src4
                   name: default.src4
             Truncated Path -> Alias:
-              /src4 [a]
+              /src4 [null-subquery2:s-subquery2:a]
         Map 6 
             Map Operator Tree:
                 TableScan
@@ -1571,7 +1571,7 @@ STAGE PLANS:
                     name: default.src5
                   name: default.src5
             Truncated Path -> Alias:
-              /src5 [b]
+              /src5 [null-subquery2:s-subquery2:b]
         Reducer 4 
             Needs Tagging: true
             Reduce Operator Tree:
diff --git a/ql/src/test/results/clientpositive/spark/union_ppr.q.out b/ql/src/test/results/clientpositive/spark/union_ppr.q.out
index 5fad0ed..4209c67 100644
--- a/ql/src/test/results/clientpositive/spark/union_ppr.q.out
+++ b/ql/src/test/results/clientpositive/spark/union_ppr.q.out
@@ -235,8 +235,8 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [x]
-              /srcpart/ds=2008-04-08/hr=12 [x]
+              /srcpart/ds=2008-04-08/hr=11 [null-subquery1:a-subquery1:x]
+              /srcpart/ds=2008-04-08/hr=12 [null-subquery1:a-subquery1:x]
         Map 3 
             Map Operator Tree:
                 TableScan
@@ -357,8 +357,8 @@ STAGE PLANS:
                     name: default.srcpart
                   name: default.srcpart
             Truncated Path -> Alias:
-              /srcpart/ds=2008-04-08/hr=11 [y]
-              /srcpart/ds=2008-04-08/hr=12 [y]
+              /srcpart/ds=2008-04-08/hr=11 [null-subquery2:a-subquery2:y]
+              /srcpart/ds=2008-04-08/hr=12 [null-subquery2:a-subquery2:y]
         Reducer 2 
             Needs Tagging: false
             Reduce Operator Tree:
-- 
1.7.9.5

